docusaurus/docs/how-to-guides/data-source-plugins/add-support-for-pdc.md

```diff
@@ -0,0 +1,220 @@
+
+- `HTTPClientOptions(ctx)` reads and creates an HTTP client configuration based on the context passed from the Grafana process.
+- `httpclient.New(opts)` calls `GetTransport()`. [Transport](https://github.com/grafana/grafana-plugin-sdk-go/blob/main/backend/httpclient/http_client.go#L90-L94) object is the one responsible for handling TLS, proxies, and other configurations within [the standard package](https://pkg.go.dev/net/http#hdr-Clients_and_Transports).
+- `GetTransport() uses ConfigureSecureSocksHTTPProxy()` to wrap the `Transport` object into socks5 proxy with TLS
+- `ConfigureSecureSocksHTTPProxy()` calls `NewSecureSocksProxyContextDialer()` which creates a socks proxy dialer.
+
+This will make every request go first through the proxy (and therefore, the PDC agent), and then reach the data source.
+
+```go
+func NewDatasource(ctx context.Context, s backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+    opts, err := s.HTTPClientOptions(ctx)
+    if err != nil {
+        return nil, err
+    }
+    httpClient, err := httpclient.New(opts)
+    if err != nil {
+        return nil, err
+    }
+    return &Datasource{HTTPClient: httpClient}, nil
+}
+
+```
+
+Here are a few examples of how it was done for some data sources:
+
+- [BigQuery](https://github.com/grafana/google-bigquery-datasource/pull/193/) \- an external library was used to connect to datasource in those cases, but it allowed swapping the default HTTP client. So all we had to do to configure the PDC proxy was to pass a client from the SDK.
+- [VictoriaMetrics Metrics Datasource](https://github.com/VictoriaMetrics/victoriametrics-datasource/releases/tag/v0.15.1)
+
+### Non-HTTP client
+
+We have a couple of options here, depending on what is available in the data source codebase:
+
+#### Override the Dialer
+
+Some packages expose means to set the [Dialer](https://pkg.go.dev/golang.org/x/net/proxy#Dialer) for the client transport. Set the `dialer` in a client with the one returned by `NewSecureSocksProxyContextDialer`.
+
+Overriding the dialer is less risky than overriding the whole Transport, as we’re changing fewer things.
+
+For example, our PostgreSQL client allows replacing the default dialer with one that supports secure SOCKS proxying ([link to the code](https://github.com/grafana/grafana/blob/da24ad06bd90b6caeaa7ad553e0063f62b0b6c5c/pkg/tsdb/grafana-postgresql-datasource/postgres.go#L71-L80)):
+
+```go
+if proxyClient.SecureSocksProxyEnabled() {
+socksDialer, err := proxyClient.NewSecureSocksProxyContextDialer()
+if err != nil {
+logger.Error("postgres proxy creation failed", "error", err)
+return nil, nil, fmt.Errorf("postgres proxy creation failed")
+}
+
+d := newPostgresProxyDialer(socksDialer)
+
+// update the postgres dialer with the proxy dialer
+connector.Dialer(d)
+}
+```
```

Comment:
I don't think this is indented?

Reply:
Yep, done https://github.com/grafana/plugin-tools/pull/1800/commits/07d9a315a6a37023967f26ae6df1b667ae1dab1e